math/automorphic/isautomorphic.go

```diff
@@ -0,0 +1,35 @@
+// isautomorphic.go
+// description: Checks whether a whole number integer is Automorphic or not. If number < 0 then returns false.
+// details:
+// In mathematics, a number n is said to be a Automorphic number if the square of n ends in the same digits as n itself.
+// ref: (https://en.wikipedia.org/wiki/Automorphic_number)
+// time complexity: O(log10(N))
+// space complexity: O(1)
+// author: [SilverDragonOfR](https://github.com/SilverDragonOfR)
+
+// Package sumofdigits provides method to determine whether a number is automorphic or not.
+package automorphic
+
+func IsAutomorphic(n int) bool {
```

Comment:
Can you think of other types that this might be useful for? Do you think you can use [type parameters](https://go.dev/tour/generics/1) here? (Hint: we have a `constraints` package in this repository)

Reply:
Thanks for informing about `contraints` package. I will use 'Integer' for n ( it makes no sense for for Float ).